pkg/cmd/server/bootstrappolicy/policy.go

```diff
@@ -156,8 +156,20 @@ func GetBootstrapClusterRoles() []authorizationapi.ClusterRole {
 			},
 			Rules: []authorizationapi.PolicyRule{
 				{
-					Verbs:     util.NewStringSet(authorizationapi.VerbAll),
-					Resources: util.NewStringSet(authorizationapi.ResourceAll),
+					// replicationControllerGetter
+					Verbs:     util.NewStringSet("get", "list"),
+					Resources: util.NewStringSet("replicationcontrollers"),
+				},
+				{
+					// RecreateDeploymentStrategy.replicationControllerClient
+					// RollingDeploymentStrategy.updaterClient
+					Verbs:     util.NewStringSet("get", "update"),
+					Resources: util.NewStringSet("replicationcontrollers"),
+				},
+				{
+					// RecreateDeploymentStrategy.hookExecutor
+					Verbs:     util.NewStringSet("get", "create"),
```

Comment:
Create is really dangerous for deployers in the long run.

----- Original Message -----
> > @@ -156,8 +156,20 @@ func GetBootstrapClusterRoles()
> > []authorizationapi.ClusterRole {
> >  			},
> >  			Rules: []authorizationapi.PolicyRule{
> >  				{
> > -					Verbs:     util.NewStringSet(authorizationapi.VerbAll),
> > -					Resources: util.NewStringSet(authorizationapi.ResourceAll),
> > +					// replicationControllerGetter
> > +					Verbs:     util.NewStringSet("get", "list"),
> > +					Resources: util.NewStringSet("replicationcontrollers"),
> > +				},
> > +				{
> > +					// RecreateDeploymentStrategy.replicationControllerClient
> > +					// RollingDeploymentStrategy.updaterClient
> > +					Verbs:     util.NewStringSet("get", "update"),
> > +					Resources: util.NewStringSet("replicationcontrollers"),
> > +				},
> > +				{
> > +					// RecreateDeploymentStrategy.hookExecutor
> > +					Verbs:     util.NewStringSet("get", "create"),
> 
> The deployer code needs `get`, `create`, and `watch` for pods.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/2477/files#r31062742